### PR TITLE
Address many long-standing LGTM complaints

### LINKED ISSUE #8320: Address many long-standing LGTM complaints
**Describe the bug**
LGTM calls out many things that it sees as problems, which we have never addressed.

**To Reproduce**
Submit a pull request, and watch LGTM report more than 150 issues that have nothing to do with your changes.

**Expected behavior**
LGTM should have little or (preferably) nothing to point out.

### PR DESCRIPTION
**Pull request by mwoodiupui**
## References
* Related to #8320

## Description
This is a first pass at addressing the backlog of LGTM alerts.  It is not complete, but has grown to a size which suggests that it should be proposed for merger now and further work moved to a new PR.

## Instructions for Reviewers
There are too many individual changes to list.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new, third-party dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies the REST API, I've linked to the REST Contract page (or open PR) related to this change.

### CODE REVIEW

**Comment by tdonohue** [comment]
@mwoodiupui : Wow! [101 alerts already fixed](https://github.com/DSpace/DSpace/pull/8321#issuecomment-1183557940).   Just wanted to note that, rather than trying to strive for "perfection", you might want to consider what is a "reasonable stopping point" for this PR.  I'd hate for this to wait for too long, or wait so long that it grows and grows (in terms of lines of code changed).  

I'll leave it up to you when you want to move this out of "draft" status. Just wanted to note that we may not want this to turn into a several thousand line PR, as the larger it gets, the more complex the review/testing process may be.

File: dspace-sword/src/main/java/org/purl/sword/client/ClientOptions.java [hunk 1]
```diff
@@ -148,128 +155,122 @@ public class ClientOptions {
...
-                    filetype = args[i];
-                }
+        option = Option.builder().longOpt("formatNamespace").hasArg().build();
+        options.addOption(option);
 
-                if ("-slug".equals(args[i])) {
-                    i++;
-                    slug = args[i];
-                }
+        option = Option.builder().longOpt("checksumError").hasArg().build();
```
Comment (tdonohue) [inline dspace-sword/src/main/java/org/purl/sword/client/ClientOptions.java:202]: In the original code, this doesn't appear to have an argument, so I think the `hasArg()` should be dropped here.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@mwoodiupui : Overall, this looks great!  I did a thorough code review today and found a few small, accidental bugs in a few new command-line options.  Everything else looks good to me... and I did a side-by-side comparison of old code vs new & referenced LGTM if I wondered why you made a specific change.

So, I think once these minor things are cleaned up, this can likely be merged. I don't see any other issues here.

File: dspace-sword/src/main/java/org/purl/sword/client/ClientOptions.java (hunk 1 above)
Comment (mwoodiupui) [reply dspace-sword/src/main/java/org/purl/sword/client/ClientOptions.java:202]: Correct, thanks!

File: dspace-api/src/main/java/org/dspace/app/statistics/LogAnalyser.java [hunk 2]
```diff
@@ -337,44 +342,72 @@ public static void main(String[] argv)
...
+        option = Option.builder().longOpt("start").hasArg().build();
+        options.addOption(option);
 
-            if (argv[i].equals("-lookup")) {
-                myLookUp = true;
-            }
+        option = Option.builder().longOpt("end").hasArg().build();
+        options.addOption(option);
+
+        option = Option.builder().longOpt("lookup").build();
```
[the remark this answers is not in the document]
Comment (mwoodiupui) [reply dspace-api/src/main/java/org/dspace/app/statistics/LogAnalyser.java:370]: Thanks for catching that!

**Review by tdonohue — APPROVED** [review]
👍 Thanks @mwoodiupui !  This looks good to me now.  I also pulled down the branch and did some light testing (to verify no obvious issues)...and no issues were found.

I'll merge this immediately since this was previously flagged as `1 approval`.

**Pull request merged**